### PR TITLE
IAM 922

### PR DESCRIPTION
**Pull request by shipperizer**
IAM-922: skaffold setup for oidc authentication

to be rebased on top of #360 once merged

- **ci: point oidc to iam.dev.canonical.com**
- **ci: use canonical hydra**
- **ci: add init container and pvc to support third party certificates inside the rock**
- **docs: add OIDC setup**
- **fix: remove login component from ui**

@huwshimi @vladimir-cucu please have a look at https://github.com/canonical/identity-platform-admin-ui/pull/361/commits/b08a9185b5f7bd3f4ae3e8fbe0666dbbe844406c (and commit message with extra description)
let me know if that's fine

Screenshot from 2024-07-18 10-49-14
Screenshot from 2024-07-18 10-49-50

### CODE REVIEW

**Comment by huwshimi** [comment]
> @huwshimi @vladimir-cucu please have a look at [b08a918](https://github.com/canonical/identity-platform-admin-ui/commit/b08a9185b5f7bd3f4ae3e8fbe0666dbbe844406c) (and commit message with extra description) let me know if that's fine

That commit looks OK, the requests won't include the `X-Authorization` header until we've implemented the new OIDC flow.

**Comment by shipperizer** [comment]
> > @huwshimi @vladimir-cucu please have a look at [b08a918](https://github.com/canonical/identity-platform-admin-ui/commit/b08a9185b5f7bd3f4ae3e8fbe0666dbbe844406c) (and commit message with extra description) let me know if that's fine
> 
> That commit looks OK, the requests won't include the `X-Authorization` header until we've implemented the new OIDC flow.

that is completely fine, `X-Authorization` header was only used as a stop gap solution until we had the authn piece wired up, which has been done

so as long as u configure the OIDC env vars and when u want to use the ui u hit `/api/v0/auth`  beforehand, all will wokr as expected (your user email will be the authorization identifier that OpenFGA will check agianst its tuples so a new seeding will be needed)

File: DEVSETUP.md
Comment (nsklikas) [inline DEVSETUP.md:151]: Can we populate the configmap from env vars (https://skaffold.dev/docs/environment/templating/)? Having to edit a committed file can be a little annoying.

Not sure if this will work

Comment (shipperizer) [reply DEVSETUP.md:151]: @nsklikas spent a bit of time on this, so long story short there is not a quick way

to get it working we should mvoe our manifests to use helm, create some values file and then we should be able to specify those via skaffold which would source them from the env

reason is only some of the skaffold fields are templatable, and those are mainly helm related see https://skaffold.dev/docs/environment/templating/

`skaffold.env` doesn't do the trick, not even via kustomize which is what i tried before, i ll poke around another bit and let you know but as it stays not sure i have a quick fix

**Review by BarcoMasile — APPROVED** [review]

**Comment by BarcoMasile** [comment]
I'd say let's remove automerge and wait for final approval by Huw for the UI changes

File: DEVSETUP.md
Comment (huwshimi) [inline DEVSETUP.md:153]: Is there any way around this limitation? This prevents us from running the service inside a container (changing OAUTH2_REDIRECT_URI gives `The 'redirect_uri' parameter does not match any of the OAuth 2.0 Client's pre-registered redirect urls.`).

Comment (huwshimi) [reply DEVSETUP.md:153]: As suggested by @shipperizer using a port forward resolves this.

**Review by huwshimi — APPROVED** [review]
This is working for me. There are a couple of pieces of redundant UI code (x-auth headers and login component) but they can be removed as part of the OIDC implementation.

**Pull request merged**
